scenario/publishsheet.go

```diff
@@ -64,7 +65,24 @@ func (publishSheetSettings PublishSheetSettings) Execute(sessionState *session.S
 	actionState *action.State, connectionSettings *connection.ConnectionSettings, label string, reset func()) {
 
 	publishAction := func(sheet *senseobjects.Sheet, ctx context.Context) error {
-		return sheet.GenericObject.Publish(ctx)
+		if publishSheetSettings.IgnorePublished {
+			if sheet.Layout.Meta.Published {
+				sessionState.LogDebugf(`not publishing sheet<%s> "%s" since it is already published`, sheet.ID, sheet.Layout.Meta.Title)
+				return nil
+			}
+		}
+		sheetAccessLevel := "private"
+		if sheet.Layout.Meta.Published {
+			sheetAccessLevel = "public"
+		}
+
+		sessionState.LogDebugf(`publishing %s sheet<%s> "%s"`, sheetAccessLevel, sheet.ID, sheet.Layout.Meta.Title)
+
+		return errors.Wrapf(
+			sheet.GenericObject.Publish(ctx),
```

Comment:
Kind of don't like this wrap-or-skip-if-nil, since it makes the success case a bit harder to read.

Reply:
maybe separate it into two lines,

err := sheet.GenericObject.Publish(ctx)

return errors.Wrapf(err, ...

Reply:
Alrighty!